### PR TITLE
Simplify object spec pre-processing

### PR DESCRIPTION
**Pull request by ash2k**
No functional changes.

### CODE REVIEW

File: pkg/controller/bundlec/deployment.go [hunk 1]
```diff
@@ -19,17 +18,7 @@ const (
 )
 
 // works around https://github.com/kubernetes/kubernetes/issues/22368
```
Comment (nilebox) [inline pkg/controller/bundlec/deployment.go]: Comment should be moved/deleted?

Comment (nilebox) [reply pkg/controller/bundlec/deployment.go]: nvm, all good.

File: pkg/controller/bundlec/resource_sync_task.go [hunk 2]
```diff
@@ -497,6 +479,20 @@ func (st *resourceSyncTask) evalSpec(res *smith_v1.Resource, actual runtime.Obje
 	}
 	obj.SetOwnerReferences(refs)
 
+	// Object GroupKind-specific munging
+	switch obj.GroupVersionKind().GroupKind() {
+	case schema.GroupKind{Group: sc_v1b1.GroupName, Kind: "ServiceInstance"}:
```
Comment (nilebox) [inline pkg/controller/bundlec/resource_sync_task.go]: nit: should Smith have constants for all supported Kinds?

File: pkg/controller/bundlec/resource_sync_task.go [hunk 3]
```diff
@@ -497,6 +479,20 @@ func (st *resourceSyncTask) evalSpec(res *smith_v1.Resource, actual runtime.Obje
...
+		obj, err = st.processServiceInstance(obj, actual, st.bundle.Namespace)
```
Comment (nilebox) [inline pkg/controller/bundlec/resource_sync_task.go]: Could just return there (and for deployment as well). Fine if you prefer to explicitly check error.

File: pkg/controller/bundlec/resource_sync_task.go (hunk 2 above)
Comment (ash2k) [reply pkg/controller/bundlec/resource_sync_task.go]: Yes, would be good to do this across the whole codebase.

File: pkg/controller/bundlec/resource_sync_task.go (hunk 3 above)
Comment (ash2k) [reply pkg/controller/bundlec/resource_sync_task.go]: Yes, I prefer that. It follows the style.

**Pull request merged**